Makefile

```diff
@@ -18,7 +18,11 @@ check:
 
 .PHONY: test
 test: fmt lint
-	cargo test --workspace
+	cargo test --workspace --lib
```

Comment:
what about leaving the `test` target to run all the tests, and introduce a new `unit-tests` target?